### PR TITLE
feat: private llm

### PR DESCRIPTION
**Pull request by mattzcarey**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context.

Uses private LLM

## Checklist before requesting a review

Please delete options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented hard-to-understand areas
- [ ] I have ideally added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: backend/llm/brainpicking.py [hunk 1]
```diff
@@ -59,21 +65,22 @@ class BrainPicking(BaseModel):
     Main class for the Brain Picking functionality.
     It allows to initialize a Chat model, generate questions and retrieve answers using ConversationalRetrievalChain.
     """
-    
+
     # Default class attributes
     llm_name: str = "gpt-3.5-turbo"
     settings = BrainSettings()
+    llm_config = LLMSettings()
```
Comment (gozineb) [inline backend/llm/brainpicking.py:72]: config or settings ?

File: docs/docs/backend/llm/private-llm.md
Comment (gozineb) [inline docs/docs/backend/llm/private-llm.md:1]: GREAT DOC !!

**Review by gozineb — APPROVED** [review]
Great Job @mattzcarey  !! Very low [logaf](https://blog.danlew.net/2020/04/15/the-logaf-scale/) on my questions/remarks

File: backend/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.11-buster
```
[the remark this answers is not in the document]
Comment (mattzcarey) [reply backend/Dockerfile:1]: The version is cmake is too low to build gpt4all.

File: backend/llm/brainpicking.py (hunk 1 above)
Comment (mattzcarey) [reply backend/llm/brainpicking.py:72]: Hmm they are the same thing in my mind. Config is normally an object. Settings are the individual attributes.

**Comment by mattzcarey** [comment]
@mamadoudicko @StanGirard

File: backend/vectorstore/supabase.py [hunk 3]
```diff
@@ -1,28 +1,21 @@
...
 from langchain.vectorstores import SupabaseVectorStore
-from supabase import Client, create_client
+from supabase import Client
 
 
 class CustomSupabaseVectorStore(SupabaseVectorStore):
     '''A custom vector store that uses the match_vectors table instead of the vectors table.'''
     user_id: str
-    def __init__(self, client: Client, embedding: OpenAIEmbeddings, table_name: str, user_id: str = "none"):
+    def __init__(self, client: Client, embedding: Embeddings, table_name: str, user_id: str = "none"):
```
Comment (mattzcarey) [inline backend/vectorstore/supabase.py:12]:
```suggestion
    def __init__(self, client: Client, embedding: OpenAIEmbeddings, table_name: str, user_id: str = "none"):
```

File: backend/vectorstore/supabase.py [hunk 4]
```diff
@@ -1,28 +1,21 @@
...
 
-from langchain.chains import ConversationalRetrievalChain, LLMChain
-from langchain.chains.question_answering import load_qa_chain
-from langchain.chat_models import ChatOpenAI, ChatVertexAI
-from langchain.client import arun_on_dataset
 from langchain.docstore.document import Document
-from langchain.embeddings.openai import OpenAIEmbeddings
-from langchain.llms import OpenAI, VertexAI
-from langchain.prompts.prompt import PromptTemplate
+from langchain.embeddings.base import Embeddings
```
Comment (mattzcarey) [inline backend/vectorstore/supabase.py:4]:
```suggestion
from langchain.embeddings.openai import OpenAIEmbeddings
```

**Review by mamadoudicko — APPROVED** [review]

**Pull request merged**